src/scripts/post_process_externally_managed_content.py

```diff
@@ -0,0 +1,112 @@
+import click
+import os
+import pandas as pd
+import logging
+
+logger = logging.getLogger(__name__)
+
+def is_qc_failure_related_to_external_source(qc_failure, erroneous_row, external):
+    # We already know the subject is the same as the mondo_id in the record
+    # Now we only need to ensure that the qc failure is related to that specific external content
+    
+    columns = [] 
+    
+    if external == "nord":
+        columns = ["report_ref", "preferred_name", "subset"]
+    elif external == "mondo-otar-subset":
+        columns = ["subset"]
+    elif external == "mondo-omim-genes":
+        columns = ["hgnc_id"]
+    elif external == "mondo-clingen":
+        columns = ["synonym", "subset"]
+    elif external == "mondo-medgen":
+        columns = ["xref_id"]
+    elif external == "mondo-efo":
+        columns = ["xref"]
+    elif external == "nando-mappings":
+        columns = ["object_id"]
+    elif external == "ordo-subsets":
+        columns = ["subset"]
+    else:
+        raise ValueError(f"Unknown external source {external}")
+    
+    for column in columns:
+        if column in erroneous_row:
+            value = erroneous_row[column]
+            qc_value = qc_failure["Value"]
+            if value == qc_value:
+                return column
+    
+    return False
+
+def write_nice_report(report, external):
+    nice_report = f"../ontology/external/{external}-qc-failures.md"
+    print(f"Writing nice report to {nice_report}")
+    
+    df_records = pd.DataFrame(report)
+    df_records.fillna("", inplace=True)
+    
+    if report:
+        report_string = df_records.to_markdown(index=False)
+    else:
+        report_string = "No QC failures found."
+    
+    failure_report = f"""
+# QC Report for {external}
+
+{report_string}
+"""
+    
+    with open(nice_report, "w") as f:
+        f.write(failure_report)
+
+def process_externally_managed_source(source, external_content_dir, robot_report_file):
```

Comment:
## Purpose & naming
"Processed" and "post-processing' is a very general words, and since this function and script don't have docstrings, then without looking closely at the code I do not know what kind of processing happening.

Correct me if I'm wrong, but it looks like the entire purpose is to filter out erroneous cells and/or rows in ROBOT templates which would otherwise lead to QC issues in `mondo`, as mentioned on [slack](https://monarchinitiative.slack.com/archives/G01JACCEDAB/p1724105022538769).

Reply:
Yes, I changed the name now to `remove_erroneous_values_from_externally_managed_content`.. Thank you!